niftypad/models.py

```diff
@@ -247,14 +288,21 @@ def logan_ref(tac, dt, inputf1, linear_phase_start, linear_phase_end, fig):
 # logan_ref_k2p - logan reference plot with fixed k2p for tac, based on eq.6 in
 # # "Distribution Volume Ratios Without Blood Sampling from Graphical Analysis of PET Data"
 
-def logan_ref_k2p(tac, dt, inputf1, k2p, linear_phase_start, linear_phase_end, fig):
+def logan_ref_k2p(tac, dt, inputf1, k2p, w, linear_phase_start_l2, linear_phase_end_l2, fig):
+    if w is None:
+        w = np.ones_like(tac)
+    linear_phase_start = linear_phase_start_l2
+    linear_phase_end = linear_phase_end_l2
     if linear_phase_start is None:
         linear_phase_start = 0
     if linear_phase_end is None:
         linear_phase_end = np.amax(dt)
+    if linear_phase_end > np.amax(dt):
```

Comment:
what if linear_phase_end `< np.amax(dt)`? Or `< 0`? Or even `< linear_phase_start`?

Reply:
-  linear_phase_end < np.amax(dt)
this is fine
- < 0
error
- < linear_phase_start
error

Reply:
ok so we should check for those and raise errors then, right?